packages/aws-cdk-lib/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -380,7 +382,7 @@ export class BucketDeployment extends Construct {
       lambdaPurpose: 'Custom::CDKBucketDeployment',
       timeout: cdk.Duration.minutes(15),
       role: props.role,
-      memorySize: props.memoryLimit,
+      memorySize: props.memoryLimit ?? (FeatureFlags.of(this).isEnabled(cxapi.S3_DEPLOYMENT_USE_512MB_MEMORY) ? 512 : undefined),
```

Comment:
I'm confused why this needs to be hidden under a feature flag. Changing the memory limit does not require replacement in the lambda. Yes, the template will see a diff when a user updates but I don't consider that a breaking change. a user would be able to deploy without replacement and without failure in their custom resource right?